### PR TITLE
LPS-171425 - [Raylife AP] Fix sort functionality to Claims Table

### PR DESCRIPTION
**Pull request by eduardo-diniz**
[LPS-171425](https://issues.liferay.com/browse/LPS-171425)

### CODE REVIEW

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/common/components/table/index.tsx [hunk 1]
```diff
@@ -12,32 +12,37 @@
...
 type TableProps = {
 	actions: ActionObject[];
 	data: {[keys: string]: string}[];
 	headers: TableHeaders[];
 	onClickRules?: (item: any, rowContent: any) => void;
-	setSortByDate?: () => void;
+	onSaveCurrent?: (item: any) => void;
+	setSortByDate?: (item: any) => void;
+	setSortState?: (item: any) => void;
+	sort?: any;
```
Comment (joanagrego) [inline modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/common/components/table/index.tsx:30]: Please type it correctly.

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/common/components/table/index.tsx [hunk 2]
```diff
@@ -46,32 +51,79 @@ const Table: React.FC<TableProps> = ({
...
+			setSortState(newSort);
+		}
+	};
+
 	return (
 		<table className="border-0 ray-table show-quick-actions-on-hover table table-autofit table-list table-responsive">
 			<Head>
 				<Row className="ray-table-head">
-					{headers.map((header, index) => (
+					{headers.map((header, index: any) => (
```
Comment (joanagrego) [inline modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/common/components/table/index.tsx:90]: Please type it correctly.

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/common/components/table/index.tsx [hunk 3]
```diff
@@ -46,32 +51,79 @@ const Table: React.FC<TableProps> = ({
...
 }) => {
+	const setCurrentHeaderPlus = (item: string) => {
+		if (onSaveCurrent) {
+			onSaveCurrent(item);
+		}
+	};
+
+	// eslint-disable-next-line react-hooks/exhaustive-deps
+	const updateSort = (colunn: string) => {
+		const newSort: any = {};
```
Comment (joanagrego) [inline modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/common/components/table/index.tsx:69]: Please type it correctly.

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/claims/pages/ClaimsTable/index.tsx [hunk 4]
```diff
@@ -533,6 +560,11 @@ const ClaimsTable = () => {
 		sortByDate === 'desc' ? setSortByDate('asc') : setSortByDate('desc');
 	};
 
+	const setHeader = (user: any) => {
```
Comment (joanagrego) [inline modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/claims/pages/ClaimsTable/index.tsx:563]: Please type it correctly.

**Review by joanagrego — CHANGES_REQUESTED** [review]
Nice job! I added some suggestions :)

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/common/components/table/index.tsx (hunk 2 above)
Comment (eduardo-diniz) [reply modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/common/components/table/index.tsx:90]: Done!

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/claims/pages/ClaimsTable/index.tsx (hunk 4 above)
Comment (eduardo-diniz) [reply modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/claims/pages/ClaimsTable/index.tsx:563]: Done!

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/common/components/table/index.tsx (hunk 3 above)
Comment (eduardo-diniz) [reply modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/common/components/table/index.tsx:69]: done!

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/common/components/table/index.tsx [hunk 5]
```diff
@@ -12,66 +12,122 @@
...
+
+	const updateSort = (colunn: string) => {
+		const newSort: sort = {};
+
+		headers.forEach((item) => {
+			if (item.requestLabel === colunn) {
+				newSort[item.key] = true;
+
+				return;
+			}
```
Comment (silasbsi) [inline modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/common/components/table/index.tsx:83]: Please, change it.
```suggestion
			if (item.requestLabel === colunn) {
				return newSort[item.key] = true;
			}
```

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/common/components/table/index.tsx [hunk 6]
```diff
@@ -12,66 +12,122 @@
...
+
+		setSort?.(newSort);
+	};
+
 	return (
 		<table className="border-0 ray-table show-quick-actions-on-hover table table-autofit table-list table-responsive">
 			<Head>
 				<Row className="ray-table-head">
-					{headers.map((header, index) => (
+					{headers.map((header, index: number) => (
```
Comment (silasbsi) [inline modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/common/components/table/index.tsx:94]: Could you put the type to header parameter?
```suggestion
					{headers.map((header: ???, index: number) => (
```

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/claims/pages/ClaimsTable/index.tsx [hunk 7]
```diff
@@ -530,7 +561,12 @@ const ClaimsTable = () => {
 	};
 
 	const setSortRule = () => {
-		sortByDate === 'desc' ? setSortByDate('asc') : setSortByDate('desc');
+		sortedOrder === 'desc' ? setSortedOrder('asc') : setSortedOrder('desc');
```
Comment (silasbsi) [inline modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/claims/pages/ClaimsTable/index.tsx:564]: Could you use 'desc' and 'asc' as an `enum`.

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/policies/pages/PoliciesTable/index.tsx [hunk 8]
```diff
@@ -517,8 +553,7 @@ const PoliciesTable = () => {
 						}
 						if (renewalDue === 0) {
 							return 'Due Today';
-						}
-						else {
+						} else {
```
Comment (silasbsi) [inline modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/policies/pages/PoliciesTable/index.tsx:556]: Please, change it.
```suggestion
						if (renewalDue === 0) {
							return 'Due Today';
						} 
						
						return renewalDue;
						
```

**Review by silasbsi — CHANGES_REQUESTED** [review]
I added some suggestions.

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/common/components/table/index.tsx (hunk 5 above)
Comment (eduardo-diniz) [reply modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/common/components/table/index.tsx:83]: Done!

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/common/components/table/index.tsx (hunk 6 above)
Comment (eduardo-diniz) [reply modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/common/components/table/index.tsx:94]: Done!

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/policies/pages/PoliciesTable/index.tsx (hunk 8 above)
Comment (eduardo-diniz) [reply modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/policies/pages/PoliciesTable/index.tsx:556]: Done!

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/claims/pages/ClaimsTable/index.tsx (hunk 7 above)
Comment (eduardo-diniz) [reply modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/claims/pages/ClaimsTable/index.tsx]: Yes, sure.
Done!

**Review by silasbsi — APPROVED** [review]

**Review by joanagrego — APPROVED** [review]
Approved! Thank you @eduardo-diniz :white_check_mark:

**Comment by silasbsi** [comment]
ci:forward

**Comment by eduardo-diniz** [comment]
ci:test:sf

**Comment by eduardo-diniz** [comment]
ci:forward

**Comment by joanagrego** [comment]
ci:forward

**Pull request closed without merging**